src/cache.rs

```diff
@@ -407,6 +409,32 @@ impl Cache {
 
         Ok(true)
     }
+
+    fn check_for_old_custom_pages(&self, custom_pages_dir: &Path) {
+        let old_custom_pages_exist = WalkDir::new(custom_pages_dir)
+            .min_depth(1)
+            .max_depth(1)
+            .into_iter()
+            .filter_entry(|entry| entry.file_type().is_file())
+            .any(|entry| {
+                if let Ok(entry) = entry {
+                    let extension = entry.path().extension();
+                    if let Some(extension) = extension {
+                        extension == "page" || extension == "patch"
+                    } else {
+                        false
+                    }
+                } else {
+                    false
+                }
+            });
```

Comment:
This check could have been written without the `WalkDir` crate, just using `fs::read_dir`. But I started rewriting it, and noticed that the `WalkDir` API is clearly nicer, so let's stick to it 🙂